lib/Services/api.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:basic_authentication_flutter_challenge/models/person.dart';
+import 'package:http/http.dart' as http;
+
+class Authentication {
+  static String accessToken = '';
+  static String userId = '';
+  static Person person = Person();
+  static final List<Person> teamMates = [];
+
+  static Future<bool> login(String username, String password) async {
+    String credentials = "$username:$password";
+    Codec<String, String> stringToBase64 = utf8.fuse(base64);
+    String encoded = stringToBase64.encode(credentials);
+    var url = Uri.parse('http://161.35.99.225/api/v1/auth/login');
+    var response = await http.post(
+      url,
+      headers: {
+        HttpHeaders.authorizationHeader: 'Basic $encoded',
+      },
+    );
+    if (response.statusCode == 200) {
+      var decodedData = json.decode(response.body);
+      userId = decodedData['result']['id'].toString();
+      accessToken = decodedData['result']['accessToken'];
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  static Future<bool> getUser() async {
+    var url = Uri.parse('http://161.35.99.225/api/v1/users/$userId');
+    var response = await http.get(
+      url,
+      headers: {
+        HttpHeaders.authorizationHeader: accessToken,
+      },
+    );
+    if (response.statusCode == 200) {
+      var jsonResponse = jsonDecode(response.body);
+      jsonResponse = jsonResponse['result'];
+      person = Person.fromJson(jsonResponse);
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  static Future<bool> getAllUsers() async {
```

Comment:
To have a better separation of concerns, `getAllUsers` function should not be part of this class since it has nothing to do with authentication. 
This is similar comment to `teamMates` variable comment.
Either create a different class that holds this function (which is better), or just define this function inside users_screen.dart file.